frontend/src/metabase/query_builder/constants.ts

```diff
@@ -19,6 +19,8 @@ export const MODAL_TYPES = {
   PREVIEW_QUERY: "preview-query",
 } as const;
 
+export type QueryModalType = typeof MODAL_TYPES[keyof typeof MODAL_TYPES];
```

Comment:
nit: should a type should exist under `/constants`? would it be better to move this to `/types`?